nearcore/src/state_sync.rs

```diff
@@ -77,30 +87,38 @@ pub fn spawn_state_sync_dump(
                 shard_id as ShardId,
                 chain,
                 runtime,
-                client_config,
-                bucket.clone(),
+                chain_id.clone(),
+                dump_config.restart_dump_for_shards.clone().unwrap_or_default(),
+                external.clone(),
+                dump_config.iteration_delay.unwrap_or(Duration::from_secs(10)),
+                account_id.clone(),
+                keep_running.clone(),
             )));
             arbiter_handle
         })
         .collect();
 
-    Ok(Some(StateSyncDumpHandle { handles }))
+    Ok(Some(StateSyncDumpHandle { handles, keep_running }))
 }
 
 /// Holds arbiter handles controlling the lifetime of the spawned threads.
 pub struct StateSyncDumpHandle {
     pub handles: Vec<actix_rt::ArbiterHandle>,
+    keep_running: Arc<AtomicBool>,
 }
 
 impl Drop for StateSyncDumpHandle {
     fn drop(&mut self) {
+        self.keep_running.store(false, std::sync::atomic::Ordering::Relaxed);
```

Comment:
what is the intended behavior here?

If I go:

```rust
fn make_node() -> Addr<ClientActor> {
  let NearNode { client, .. } = nearcore::start_with_config(..);
  client
}

fn main() {
  let client = make_client();
  do_stuff(client);
}
```

Then the node keeps running as usual but dumping state has stopped, which seems a bit unexpected

Reply:
what would be wrong with just not changing `keep_running` on drop? Is this behavior something we explicitly want for some reason? Could make it work like the join handle returned from `std::thread::spawn()` where dropping it does nothing, and just means that you dont have a handle to it anymore

Reply:
The problem I'm solving is that the loop isn't interrupt-able while generating parts in
```
for part_id in parts_dumped..num_parts {
````

Tested that locally:
* Ignore `keep_running`
* Replace the for-loop with an infinite loop obtaining `part_id=0`.
* Run a localnet node that dumps state to `/tmp/`.
* Hit Ctrl+C, it stops actix actors, disables logging, but the loop keeps running, writing new files over and over.

If dumping state takes only a short time, then the dumping threads respond well to the interrupts.

`keep_running` stops that long-running loop and makes the threads interrupt-able.

Reply:
Yeah but I think the behavior you describe occurs because `StateSyncDumpHandle::stop()` doesn't set `keep_running`, not because we need it to be set in `drop()`. So if you add that, it should work as you expect it to. And honestly it feels to me like `StateSyncDumpHandle::stop()` should probably set that variable right